lib/OrdinaryDiffEqRosenbrock/src/rosenbrock_perform_step.jl

```diff
@@ -1600,66 +1600,35 @@ end
     return nothing
 end
 
-function initialize!(integrator, cache::Rosenbrock5Cache)
+function initialize!(integrator, cache::RosenbrockCache)
     integrator.kshortsize = 3
-    @unpack dense1, dense2, dense3 = cache
+    dense = cache.dense
+    dense1, dense2, dense3 = dense[1], dense[2], dense[3] 
     resize!(integrator.k, integrator.kshortsize)
     integrator.k[1] = dense1
     integrator.k[2] = dense2
     integrator.k[3] = dense3
 end
 
-@muladd function perform_step!(integrator, cache::Rosenbrock5Cache, repeat_step = false)
-    @unpack t, dt, uprev, u, f, p = integrator
-    @unpack du, du1, du2, k1, k2, k3, k4, k5, k6, k7, k8, dT, J, W, uf, tf, linsolve_tmp, jac_config, atmp, weight, stage_limiter!, step_limiter! = cache
-    @unpack a21, a31, a32, a41, a42, a43, a51, a52, a53, a54, a61, a62, a63, a64, a65, C21, C31, C32, C41, C42, C43, C51, C52, C53, C54, C61, C62, C63, C64, C65, C71, C72, C73, C74, C75, C76, C81, C82, C83, C84, C85, C86, C87, gamma, d1, d2, d3, d4, d5, c2, c3, c4, c5 = cache.tab
+@muladd function perform_step!(integrator, cache::RosenbrockCache, repeat_step = false)
```

Comment:
this should be deleted (since we already wrote this perform step above)

Reply:
But it that case I need to merge both of them right away

Reply:
@oscardssmith Should I merge them ??

Reply:
I don't think the one for Rodas4 should need any changes to work for Rodas5.

Reply:
@oscardssmith But I saw that the length of the dense vector in case of Rodas5 is 3 while in Rodas4 it's 2. Will any changes be required to be made to address that ?

Reply:
I believe that should be dealt with automatically. I made the code loop from `1:length(integrator.k)` so it should just work.